packages/flutter/lib/src/widgets/fade_in_image.dart

```diff
@@ -448,7 +448,13 @@ class _AnimatedFadeOutFadeInState extends ImplicitlyAnimatedWidgetState<_Animate
         tween: ConstantTween<double>(0),
         weight: widget.fadeInDuration.inMilliseconds.toDouble(),
       ),
-    ]));
+    ]))..addStatusListener((AnimationStatus status) {
+      if (_placeholderOpacityAnimation.isCompleted) {
```

Comment:
Yes - the implementation of `isCompleted` is `status == AnimationStatus.completed`

Reply:
Oh - sorry.

This might not be the same if we've changed the `_placeholderOpacityAnimation`, but the previous one we were listening to is still finishing.

This is meant to guard against that case, so we don't end up rebuilding unless the current animation is completed.